open-banking-accelerator/components/com.wso2.openbanking.accelerator.common/src/main/java/com/wso2/openbanking/accelerator/common/util/JWTUtils.java

```
@@ -120,4 +137,120 @@ public static boolean validateJWTSignature(String jwtString, String jwksUri, Str
+
+        byte[] publicKeyData = Base64.getDecoder().decode(publicKey);
+        X509EncodedKeySpec spec = new X509EncodedKeySpec(publicKeyData);
+        KeyFactory kf = KeyFactory.getInstance(algorithm);
+        RSAPublicKey rsapublicKey = (RSAPublicKey) kf.generatePublic(spec);
+        JWSVerifier verifier = new RSASSAVerifier(rsapublicKey);
+        return signedJWT.verify(verifier);
+
+    }
+
+    /**
+     * Validate legitimacy of JWT.
+     *
+     * @param jwtString JWT string
+     */
+    public static boolean isJWT(String jwtString) {
+
+        if (jwtString == null) {
+            return false;
+        }
+        if (StringUtils.isBlank(jwtString)) {
+            return false;
+        }
+        if (StringUtils.countMatches(jwtString, DOT_SEPARATOR) != 2) {
+            return false;
+        }
+        try {
+            //Checking whether the jwtString is jwt parsable.
+            JWTParser.parse(jwtString);
+            return true;
+        } catch (ParseException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Provided JWT is not a parsable.", e);
+            }
+            return false;
+        }
+
+    }
+
+    /**
+     * Parses the provided JWT string into a SignedJWT object.
+     *
+     * @param jwtString the JWT string to parse
+     * @return the parsed SignedJWT object
+     * @throws IllegalArgumentException if the provided token identifier is not a parsable JWT
+     * Will not throw ParseException as it is already validated by isJWT
+     */
+    public static SignedJWT getSignedJWT(String jwtString) throws ParseException {
+
+        if (isJWT(jwtString)) {
+            return SignedJWT.parse(jwtString);
+        } else {
+            if (log.isDebugEnabled()) {
+                log.debug("Provided token identifier is not a parsable JWT.");
+            }
+            throw new IllegalArgumentException("Provided token identifier is not a parsable JWT.");
```

Better to throw a custom exception we have defined.